libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -47,26 +47,28 @@ export class NotificationTemplateRepository extends BaseRepository<
     return this.mapEntity(item);
   }
 
-  async findBlueprint(idOrIdentifier: string) {
+  async findBlueprintById(id: string) {
     if (!this.blueprintOrganizationId) throw new DalException('Blueprint environment id was not found');
 
-    const isMongoId = NotificationTemplateRepository.isMongoId(idOrIdentifier);
-
-    let requestQuery: NotificationTemplateQuery;
-
-    if (isMongoId) {
-      requestQuery = {
-        _id: idOrIdentifier,
-        isBlueprint: true,
-        _organizationId: this.blueprintOrganizationId,
-      };
-    } else {
-      requestQuery = {
-        triggers: { $elemMatch: { identifier: idOrIdentifier } },
-        isBlueprint: true,
-        _organizationId: this.blueprintOrganizationId,
-      };
-    }
+    const requestQuery: NotificationTemplateQuery = {
+      isBlueprint: true,
+      _organizationId: this.blueprintOrganizationId,
+      _id: id,
+    };
+
+    const item = await this.MongooseModel.findOne(requestQuery).populate('steps.template').lean();
+
+    return this.mapEntity(item);
+  }
+
+  async findBlueprintByTriggerIdentifier(identifier: string) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** Nice, well done!‏